lib/manageiq/environment.rb

```diff
@@ -54,7 +54,11 @@ def self.update_ui_thread
     end
 
     def self.install_bundler(root = APP_ROOT)
-      system!("gem install bundler -v '#{bundler_version}' --conservative") unless ENV["GITHUB_ACTIONS"]
+      if system("which bundle", [:out, :err] => "/dev/null") && system("bundle info bundler", [:out, :err] => "/dev/null")
+        puts "== proper bundler detected... skipping bundler install =="
```

Comment:
Formatting for consistency

```suggestion
        puts "== Bundler detected... Skipping bundler install =="
```

Aside, I'm wonder if we should print anything at all. Sometimes less is more.

Reply:
Also, just checking `bundle info` really isn't enough.  The install line below checks the version of bundler we need and will install a newer bundler if we bump the base dependency, whereas now I don't think it will.

Reply:
I'm testing this.  I think you will be unable to get a non-zero RC from `bundle info bundler` if you have a bundler in use that doesn't match the requirements.

Reply:
Ok, so the testing below shows that `bundle info bundler` will only return non-zero RC if you have the bundler version installed from your fully resolved and up to date Gemfile.lock. 

```
joerafaniello@Joes-MacBook-Pro-2 manageiq % gem list bundler

*** LOCAL GEMS ***

bundler (2.3.10, default: 1.17.2)
bundler-audit (0.9.0.1)
joerafaniello@Joes-MacBook-Pro-2 manageiq % CI=true SKIP_TEST_RESET=true SKIP_DATABASE_SETUP=true SKIP_UI_UPDATE=true bin/setup
== Installing dependencies ==
XXX just for testing, bundler install skipped
```


```
joerafaniello@Joes-MacBook-Pro-2 manageiq % gem uninstall bundler
Successfully uninstalled bundler-2.3.10
joerafaniello@Joes-MacBook-Pro-2 manageiq % gem list bundler

*** LOCAL GEMS ***

bundler (default: 1.17.2)
bundler-audit (0.9.0.1)
joerafaniello@Joes-MacBook-Pro-2 manageiq % bundle info bundler
/Users/joerafaniello/.rubies/ruby-2.6.9/lib/ruby/2.6.0/rubygems.rb:283:in `find_spec_for_exe': Could not find 'bundler' (2.3.10) required by your /Users/joerafaniello/Code/manageiq/Gemfile.lock. (Gem::GemNotFoundException)
To update to the latest version installed on your system, run `bundle update --bundler`.
To install the missing version, run `gem install bundler:2.3.10`
	from /Users/joerafaniello/.rubies/ruby-2.6.9/lib/ruby/2.6.0/rubygems.rb:302:in `activate_bin_path'
	from /Users/joerafaniello/.gem/ruby/2.6.9/bin/bundle:23:in `<main>'
joerafaniello@Joes-MacBook-Pro-2 manageiq % rm Gemfile.lock
joerafaniello@Joes-MacBook-Pro-2 manageiq % CI=true SKIP_TEST_RESET=true SKIP_DATABASE_SETUP=true SKIP_UI_UPDATE=true bin/setup
== Installing dependencies ==
XXX == Bundler not detected... installing ==
Fetching bundler-2.3.10.gem
Successfully installed bundler-2.3.10
1 gem installed
```

Reply:
I also tried this scenario:
1) Use bundler 2.3.10 to bundle things
2) Uninstall bundler 2.3.10
3) Install another bundler that meets the bundle's bundle requirements: 2.1.4
4) Run bin/setup

This correctly skips the bundler install because bundler is a valid version.

```
joerafaniello@Joes-MacBook-Pro-2 manageiq % grep -iC 1 bundle Gemfile.lock
      aws-sdk-s3 (~> 1.0)
      bundler (~> 2.1, >= 2.1.4, != 2.2.10)
      erubis (= 2.7.0)
--
--
    builder (3.2.4)
    bundler-audit (0.9.0.1)
      bundler (>= 1.2.0, < 3)
      thor (~> 1.0)
--
--
      activesupport (= 6.0.4.7)
      bundler (>= 1.3.0)
      railties (= 6.0.4.7)
--
--
  brakeman (~> 5.0)
  bundler (~> 2.1, >= 2.1.4, != 2.2.10)
  bundler-audit
  byebug
--
--

BUNDLED WITH
   2.3.10. # <---------- I bundled with this version
joerafaniello@Joes-MacBook-Pro-2 manageiq % gem list bundler

*** LOCAL GEMS ***

bundler (2.1.4, default: 1.17.2) # <---- I removed 2.3.10 and installed a valid 2.1.4
bundler-audit (0.9.0.1)
joerafaniello@Joes-MacBook-Pro-2 manageiq % CI=true SKIP_TEST_RESET=true SKIP_DATABASE_SETUP=true SKIP_UI_UPDATE=true bin/setup
== Installing dependencies ==
XXX just for testing, bundler install skipped
== Bundle up to data... Skipping update ==
```

Reply:
> Aside, I'm wonder if we should print anything at all. Sometimes less is more.

I think you're right.  It prints when it installs bundler automatically... if bundler is already there, who cares, we only care if it did something.